src/core/lib/event_engine/iomgr_engine/iomgr_engine.cc

```diff
@@ -67,10 +54,29 @@ std::string HandleToString(EventEngine::TaskHandle handle) {
 
 }  // namespace
 
+struct IomgrEventEngine::ClosureData final : public EventEngine::Closure {
```

Comment:
"ClosureData is a Closure" is an odd turn of phrase. I'd recommend against inheritance here.

Reply:
Not using inheritance here would lead to some relatively nasty casting that I'd like to avoid.
I'd be more ok with making TimerManager take a different type (`EventEngine::Closure` was chosen for convenience, but I'm not sure that we're getting that)